speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SocketBlackBox.h

```diff
@@ -6,18 +6,29 @@
 #include "IDecision.h"
 #include "msgpack.hpp"
 #include "DecisionTuple.h"
+#include "ServerSocket.h"
+
+#define SBB_BUFFER_SIZE 512
 
 template <class DriveSituation>
 class SocketBlackBox
 {
 public:
+
+    SocketBlackBox(PCWSTR p_ip = L"127.0.0.1", int p_port = 8888) : m_server(p_ip, p_port)
+    {
+	    
+    }
     /**
      * Initialize a socket black box:
      * connect to the black box
      * register what variables should be shared
      * etc...
      */
+
     void Initialize();
+
+    void Initialize(DriveSituation& p_initialDriveSituation, DriveSituation* p_tests = nullptr, int p_amountOfTests = 1);
```

Comment:
Might be better to set the default value of p_amountOfTests to 0. Now you will enter the test loop, but there's no DriveSituations to test (because it is a nullptr)